ParProcCo/job_slicer_interface.py

```diff
@@ -0,0 +1,58 @@
+from __future__ import annotations
+
+from copy import deepcopy
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from ParProcCo.utils import (
+    check_jobscript_is_readable,
+    check_location,
+    get_absolute_path,
+)
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from .job_scheduling_information import JobSchedulingInformation
+
+
+class JobSlicerInterface:
+    def __init__(self, job_script: Path | None = None) -> None:
+        if job_script is not None:
+            self.job_script = check_jobscript_is_readable(
+                check_location(get_absolute_path(job_script))
+            )
+        else:
+            self.job_script = Path("n/a")
```

Comment:
What? Shouldn't this be `None`? Surely `Path("n/a")` is invalid?

Reply:
Other code checks if the path is readable

Reply:
But it could be valid as a relative path, if file "a" was made in directory "n", so isn't that a security issue?

Reply:
I guess... To me it still seems like any errors that come of it would be clearer if it was set to `None`, in case `__init__()` isn't changed when creating a subclass.